linkerd/meshtls/src/client.rs

```diff
@@ -128,22 +130,29 @@ impl<I> Future for ConnectFuture<I>
 where
     I: io::AsyncRead + io::AsyncWrite + Unpin,
 {
-    type Output = io::Result<ClientIo<I>>;
+    type Output = io::Result<(ClientIo<I>, Option<NegotiatedProtocol>)>;
 
     #[inline]
     fn poll(self: std::pin::Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         match self.project() {
             #[cfg(feature = "boring")]
             ConnectFutureProj::Boring(f) => {
                 let res = futures::ready!(f.poll(cx));
-                Poll::Ready(res.map(ClientIo::Boring))
+                Poll::Ready(res.map(|io| {
+                    let np = io.negotiated_protocol().map(|np| np.to_owned());
```

Comment:
huh, i was surprised that `std` doesn't have an `impl<T: ToOwned> ToOwned for Option<T>` but i guess it would conflict with [the blanket impl for `T: Clone`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#impl-ToOwned)...